### ISSUE TITLE
Alerts are sent multiple times

### ISSUE THREAD

**Issue by papjul** [breezy-weather/breezy-weather#73]
### Steps to reproduce

1. Add "current position"
2. Add the city of this current position with the same provider
3. Wait for an alert

### Expected behavior

Alert should be received as notification only once

### Actual behavior

Alert is sent twice, once for each location

### Weather provider used

All

### Breezy Weather version

4.0.0-alpha

### Android version

All

### Device

All

### Other details

This is because the algorithm is per location and only checks if alert was part of the previous update. If it was, it won't send.

We could experiment a workaround by getting all alert list from all location of the same provider (not possible to do this across different weather providers because of different alert IDs used). This requires to improve alert IDs for all providers. Some don't have it, so we should make a hash or something like that. Current ID is a `long`, we should make it a `string`.

Also reported in Geometric Weather as WangDaYeeeeee/GeometricWeather#261, WangDaYeeeeee/GeometricWeather#437, WangDaYeeeeee/GeometricWeather#467

A perfectible workaround is to avoid having both current location and your city as different locations. Choose only one of them.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by papjul** [comment]
So, after a bit of investigation:
- Alerts are only sent for the first location in the list, so my first analysis was wrong
- There may be multiple similar alerts, example:
    - Thunderstorm yellow from 12:00 to 15:00
    - Thunderstorm orange from 15:00 to 18:00
    - Thunderstorm yellow from 18:00 to 21:00
    - => What should the app do in that case? I would say to only show alerts in advance, however that means 3 notifications for the same kind of event ;)
- It seems notifications are sent in loop when widgets are added, because widgets seem to force refreshing but it may end up in a loop (to be investigated)

**Comment by papjul** [comment]
Bug fixed with da1d4783adc9a22097dac9f12e998141617d5163

Tracking issues with AccuWeather duplicate alerts in #159
